### DISCUSSION TITLE
consult-register: how to sort items by updating time

### DISCUSSION THREAD

**Discussion by uqix** [minad/consult#1076] [Q&A]
By default ,they are sorted by alphanumeric, while in `*Register Preview*` they're sorted by updating time.

Context:
* vertico is installed

**Comment by minad** [comment]
Sorting by recency is currently not supported. I wonder if Emacs register write and read (!) commands should reorder the list, which would make such an LRU listing more useful.

**Comment by uqix** [comment]
I wrote a function to auto name (e.g. ABCABCA...) registers and list them by updating time:

```elisp
(defvar my/register/point-name ?A)

(defun my/register ()
  (interactive)
  (if (use-region-p)
      (call-interactively 'consult-register-store)
    (if (>= my/register/point-name (+ ?A 15))
        (setq my/register/point-name ?A))
    (assq-delete-all my/register/point-name register-alist)
    (point-to-register my/register/point-name)
    (message (format "Register %c" my/register/point-name))
    (cl-incf my/register/point-name)))

(defun consult-register--alist (&optional noerror filter)
  "Return register list, sorted and filtered with FILTER.
Raise an error if the list is empty and NOERROR is nil."
  (or (cl-loop for reg in register-alist
               ;; Sometimes, registers are made without a `cdr' or with
               ;; invalid markers.  Such registers don't do anything, and
               ;; can be ignored.
               if (and (cdr reg)
                       (or (not (markerp (cdr reg))) (marker-buffer (cdr reg)))
                       (or (not filter) (funcall filter reg)))
               collect reg)
      (and (not noerror) (user-error "All registers are empty"))))
```

but without `(sort ... #'car-less-than-car)` in `consult-register--alist`.

**Reply by uqix** [reply]
Maybe we can add an option to enable/disable this sorting?

**Reply by minad** [reply]
> Maybe we can add an option to enable/disable this sorting?

My opinion is still that this feature is not sufficiently useful currently. Users who want this feature can use an advice. Maybe propose adding an LRU reordering option to Emacs register.el, such that both read and write reorder the registers? If this is supported there, Consult can also support this.
